### ISSUE TITLE
POST /api/People fails with 500 error on prealpha.rocksolidchurch.com

### ISSUE THREAD

**Issue by shawnpyle** [SparkDevNetwork/Rock#5494]
### Please go through all the tasks below

- [X] Check this box only after you have successfully completed both the above tasks

### Please provide a brief description of the problem. Please do not forget to attach the relevant screenshots from your side.

After assigning the `Presence Presence` user full rights to `RSR - Rock Administration` group, a POST to https://prealpha.rocksolidchurchdemo.com/api/People fails with a 500 error.

### Expected Behavior

POST to /People on prealpha demo site should operate the same as rock demo site.

### Actual Behavior

```
zsh> curl -v -X POST \
  'https://prealpha.rocksolidchurchdemo.com/api/People' \
  --header 'Authorization-Token: C5E93131DC7848B7AF9C5EA71F821ACB' \
  --header 'Content-Type: application/json' \
  --data-raw '{"Email": "test-20220514-1730@letterx.co", "LastName": "20220514-1730", "FirstName": "StrongMad", "ConnectionStatusValueId": 0}'
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 144.202.63.130:443...
* Connected to prealpha.rocksolidchurchdemo.com (144.202.63.130) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* [CONN-0-0][CF-SSL] (304) (OUT), TLS handshake, Client hello (1):
* [CONN-0-0][CF-SSL] (304) (IN), TLS handshake, Server hello (2):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS handshake, Certificate (11):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS handshake, Server finished (14):
* [CONN-0-0][CF-SSL] TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* [CONN-0-0][CF-SSL] TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* [CONN-0-0][CF-SSL] TLSv1.2 (OUT), TLS handshake, Finished (20):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=rockrmslatest.com
*  start date: Jun 15 20:55:16 2023 GMT
*  expire date: Sep 13 20:55:15 2023 GMT
*  subjectAltName: host "prealpha.rocksolidchurchdemo.com" matched cert's "prealpha.rocksolidchurchdemo.com"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: POST]
* h2h3 [:path: /api/People]
* h2h3 [:scheme: https]
* h2h3 [:authority: prealpha.rocksolidchurchdemo.com]
* h2h3 [user-agent: curl/7.87.0]
* h2h3 [accept: */*]
* h2h3 [authorization-token: C5E93131DC7848B7AF9C5EA71F821ACB]
* h2h3 [content-type: application/json]
* h2h3 [content-length: 127]
* Using Stream ID: 1 (easy handle 0x7fbf6c00ce00)
> POST /api/People HTTP/2
> Host: prealpha.rocksolidchurchdemo.com
> user-agent: curl/7.87.0
> accept: */*
> authorization-token: C5E93131DC7848B7AF9C5EA71F821ACB
> content-type: application/json
> content-length: 127
>
* We are completely uploaded and fine
< HTTP/2 500
< cache-control: no-cache
< pragma: no-cache
< content-type: application/json; charset=utf-8
< expires: -1
< date: Mon, 26 Jun 2023 20:01:09 GMT
< content-length: 36
<
* Connection #0 to host prealpha.rocksolidchurchdemo.com left intact
{"Message":"An error has occurred."}%
```

where as rock.solidrockchurch.com yields

```
zsh> curl -v -X POST \
  'https://rock.rocksolidchurchdemo.com/api/People' \
  --header 'Authorization-Token: C5E93131DC7848B7AF9C5EA71F821ACB' \
  --header 'Content-Type: application/json' \
  --data-raw '{"Email": "test-20220514-1730@letterx.co", "LastName": "20220514-1730", "FirstName": "StrongMad", "ConnectionStatusValueId": 0}'
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying 45.63.65.46:443...
* Connected to rock.rocksolidchurchdemo.com (45.63.65.46) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* [CONN-0-0][CF-SSL] (304) (OUT), TLS handshake, Client hello (1):
* [CONN-0-0][CF-SSL] (304) (IN), TLS handshake, Server hello (2):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS handshake, Certificate (11):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS handshake, Server finished (14):
* [CONN-0-0][CF-SSL] TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* [CONN-0-0][CF-SSL] TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* [CONN-0-0][CF-SSL] TLSv1.2 (OUT), TLS handshake, Finished (20):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* [CONN-0-0][CF-SSL] TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server accepted h2
* Server certificate:
*  subject: CN=rockrmsdemo.com
*  start date: May 23 22:41:08 2023 GMT
*  expire date: Aug 21 22:41:07 2023 GMT
*  subjectAltName: host "rock.rocksolidchurchdemo.com" matched cert's "rock.rocksolidchurchdemo.com"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: POST]
* h2h3 [:path: /api/People]
* h2h3 [:scheme: https]
* h2h3 [:authority: rock.rocksolidchurchdemo.com]
* h2h3 [user-agent: curl/7.87.0]
* h2h3 [accept: */*]
* h2h3 [authorization-token: C5E93131DC7848B7AF9C5EA71F821ACB]
* h2h3 [content-type: application/json]
* h2h3 [content-length: 127]
* Using Stream ID: 1 (easy handle 0x7fd125810600)
> POST /api/People HTTP/2
> Host: rock.rocksolidchurchdemo.com
> user-agent: curl/7.87.0
> accept: */*
> authorization-token: C5E93131DC7848B7AF9C5EA71F821ACB
> content-type: application/json
> content-length: 127
>
* We are completely uploaded and fine
< HTTP/2 200
< cache-control: no-cache
< pragma: no-cache
< content-type: application/json; charset=utf-8
< expires: -1
< date: Mon, 26 Jun 2023 20:01:57 GMT
< content-length: 2
<
* Connection #0 to host rock.rocksolidchurchdemo.com left intact
90%
```

### Steps to Reproduce

1. Add Presence user to `RSR - Rock Administration` security role if not already done. https://prealpha.rocksolidchurchdemo.com/person/3/security
2. Try executing the following cURL command:
 ```
curl -v -X POST \
  'https://prealpha.rocksolidchurchdemo.com/api/People' \
  --header 'Authorization-Token: C5E93131DC7848B7AF9C5EA71F821ACB' \
  --header 'Content-Type: application/json' \
  --data-raw '{"Email": "test-20220514-1730@letterx.co", "LastName": "20220514-1730", "FirstName": "StrongMad", "ConnectionStatusValueId": 0}'
```

### Rock Version

16.0

### Client Culture Setting

English

**Comment by jonedmiston** [comment]
The issue is the data above is passing in a value for `ConnectionStatusValueId` (0) that does not exist in the database. Not sure why/how this could have worked in the past.

**Comment by shawnpyle** [comment]
Quite right @jonedmiston. Sending the following which has a correct connection status value does work.
```
curl  -X POST \
  'https://prealpha.rocksolidchurchdemo.com/api/People' \
  --header 'Authorization-Token: C5E93131DC7848B7AF9C5EA71F821ACB' \
  --header 'Content-Type: application/json' \
  --data-raw '{
  "FirstName": "Jim",
  "LastName": "Doe",
  "Email": "jimdoe@gmail.com",
  "ConnectionStatusValueId": 1224
}'
```
Response: 
```
61
```

I would not have expected giving a bad `ConnectionStatusValueId` to cause a 500 error though.
